bigtable/client/testing/embedded_server_test_fixture.cc

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/testing/embedded_server_test_fixture.h"
+#include "bigtable/client/internal/throw_delegate.h"
+#include <thread>
+
+namespace bigtable {
+namespace testing {
+
+void EmbeddedServerTestFixture::StartServer() {
+  int port;
+  std::string server_address("[::]:0");
+  builder_.AddListeningPort(server_address, grpc::InsecureServerCredentials(),
+                            &port);
+  builder_.RegisterService(&bigtable_service_);
+  builder_.RegisterService(&admin_service_);
+  server_ = builder_.BuildAndStart();
+  address_ = "localhost:" + std::to_string(port);
+  is_server_started_ = true;
+  wait_thread_ = std::thread([this]() { server_->Wait(); });
+}
+
+void EmbeddedServerTestFixture::SetUp() {
+  if (!is_server_started_) {
```

Comment:
Consider:

```C++
  if (not is_server_started_) {
    return;
  }
 ... // rest of the code here.
```

But also: why would `is_server_started_` ever be `true` when `SetUp()` is called?  As far as I know (and this I do not know with certainty, so please tell me if I am wrong), the fixture is created before each test and then `SetUp()` is called.

Reply:
you are right, i was under the impression that fixture is created before testcase and Setup before test. removed this condition.